### PR TITLE
Fixes/dependencies

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

- [x] Distributing the app it does not include Avalonia.Build.Task.
- [x] fixes some  the NU1605 warnings.
- [x] Add buildTransitive in nuget #5210. 
- [x] Migrated from SourceLink.Create.CommandLine to Microsoft.SourceLink.GitHub

## What is the current behavior?

When deploying the app it includes Avalonia.Build.Task

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Fixed issues
#5210

### CODE REVIEW

File: samples/ControlCatalog.iOS/ControlCatalog.iOS.csproj [hunk 1]
```diff
@@ -177,10 +177,13 @@
       <Project>{d0a739b9-3c68-4ba6-a328-41606954b6bd}</Project>
       <Name>ControlCatalog</Name>
     </ProjectReference>
-    <PackageReference Include="SkiaSharp.HarfBuzz" Version="2.80.2-preview.33" />
```
Comment (rstm-sf) [inline samples/ControlCatalog.iOS/ControlCatalog.iOS.csproj:180]: As far as I remember, this version was chosen on purpose, like something related to memory leaks

Comment (rstm-sf) [reply samples/ControlCatalog.iOS/ControlCatalog.iOS.csproj:180]: Or was it done for iOS #4597

Comment (workgroupengineering) [reply samples/ControlCatalog.iOS/ControlCatalog.iOS.csproj]: thanks, I didn't remember. I revert it.

File: src/Avalonia.Base/Avalonia.Base.csproj [hunk 2]
```diff
@@ -5,9 +5,6 @@
     <RootNamespace>Avalonia</RootNamespace>
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>    
   </PropertyGroup>
-  <ItemGroup>
-    <ProjectReference Include="..\Avalonia.Build.Tasks\Avalonia.Build.Tasks.csproj"/>
```
Comment (kekekeks) [inline src/Avalonia.Base/Avalonia.Base.csproj:9]: It's required to be built by Avalonia.MicroCom and probably other assemblies in the future, so it made sense to reference it from Avalonia.Base.

Comment (workgroupengineering) [reply src/Avalonia.Base/Avalonia.Base.csproj:9]: Hi,
> It's required to be built by Avalonia.MicroCom and probably other assemblies in the future, so it made sense to reference it from Avalonia.Base.

from the tests I made Avalonia.MicroCom works correctly.

I don't understand what the purpose of Avalonia.Base dependency on Avalonia.Build.Tasks is. If it is for compiling the xaml of the Core classes, I would add Avalonia.Build.Tasks as PrivateAssets in Base.prop and buildTransitive in Avalonia.nupkg.

Tell me if I have to close this PR:

Comment (kekekeks) [reply src/Avalonia.Base/Avalonia.Base.csproj:9]: Ah, MicroCom is [also referencing](https://github.com/AvaloniaUI/Avalonia/blob/c85fa2b9977d251a31886c2534613b4730fbaeaf/src/Avalonia.MicroCom/Avalonia.MicroCom.csproj#L10-L13) the build tasks. So it should be OK to remove the reference from Avalonia.Base.

File: src/Avalonia.Themes.Default/Avalonia.Themes.Default.csproj [hunk 3]
```diff
@@ -12,10 +12,8 @@
     <ProjectReference Include="..\Avalonia.Layout\Avalonia.Layout.csproj" />
     <ProjectReference Include="..\Avalonia.Visuals\Avalonia.Visuals.csproj" />
     <ProjectReference Include="..\Avalonia.Styling\Avalonia.Styling.csproj" />
-    <AvaloniaResource Include="DefaultTheme.xaml" />
-    <AvaloniaResource Include="Accents/*.xaml" />
-    <!-- Compatibility with old apps, probably need to replace with AvaloniaResource -->
-    <EmbeddedResource Include="**/*.xaml" />
+    <AvaloniaResource Include="**/*.xaml" />
+    <AvaloniaResource Include="Assets\*" />
```
Comment (rstm-sf) [inline src/Avalonia.Themes.Default/Avalonia.Themes.Default.csproj]: This is already trying to do #5204 :)

But, only the lines above

File: packages/Avalonia/Avalonia.csproj [hunk 4]
```diff
@@ -31,21 +31,23 @@
...
     <Content Include="AvaloniaItemSchema.xaml">
       <Pack>true</Pack>
-      <PackagePath>build\</PackagePath>
+      <PackagePath>build\;buildTransitive\</PackagePath>
     </Content>
   </ItemGroup>
   <Import Project="..\..\build\SharedVersion.props" />
   <Import Project="..\..\build\NetFX.props" />
   <Import Project="..\..\build\CoreLibraries.props" />
+  <Import Project="..\..\build\SourceLink.props" />
```
Comment (rstm-sf) [inline packages/Avalonia/Avalonia.csproj:52]: It is possible that a similar condition is needed
https://github.com/AvaloniaUI/Avalonia/blob/554aaec5e5cc96c0b4318b6ed1fbf8159f442889/src/Directory.Build.props#L4

**Comment by grokys** [comment]
@kekekeks is this the last thing that we need to merge for 0.10? (fix for #5210)

**Review by danwalmsley — APPROVED** [review]
LGTM, @kekekeks also approves verbally

**Pull request merged**